code/game/objects/items/stacks/sheets/sheets.dm

```diff
@@ -14,8 +14,27 @@
 	var/point_value = 0 //turn-in value for the gulag stacker - loosely relative to its rarity.
 	///What type of wall does this sheet spawn
 	var/walltype
+	///What the sheet will shatter to if you slap the floor
+	var/obj/item/shards_to
 
 /obj/item/stack/sheet/Initialize(mapload, new_amount, merge)
 	. = ..()
 	pixel_x = rand(-4, 4)
 	pixel_y = rand(-4, 4)
+
+/**
+ * Facilitates sheets being smacked on the floor
+ *
+ * This is used for crafting by hitting the floor with items.
+ * The inital use case is glass sheets breaking in to shards when the floor is hit.
+ * Args:
+ * * user: The user that did the action
+ * * params: paramas passed in from attackby
+ */
+/obj/item/stack/sheet/proc/attacked_by_floor(mob/user, params)
```

Comment:
Are you sure `attacked_by_floor` is the most clear name for something called while attacking the floor?

Reply:
Finding a better name now.